packages/loader/container-loader/src/test/loaderLayerCompatValidation.spec.ts

```diff
@@ -44,36 +45,29 @@ type ILayerCompatSupportRequirementsOverride = Omit<
 function validateFailureProperties(
 	error: Error,
 	isGenerationCompatible: boolean,
-	layerGeneration: number,
-	layerType: "runtime" | "driver",
+	incompatibleLayerGeneration: number,
+	incompatibleLayer: FluidLayer,
 	unsupportedFeatures?: string[],
 ): boolean {
-	assert(
-		isFluidError(error) && error.errorType === FluidErrorTypes.usageError,
-		"Error should be a usageError",
-	);
-	assert.strictEqual(
-		error.errorType,
-		FluidErrorTypes.usageError,
-		"Error type should be usageError",
-	);
-	const telemetryProps = error.getTelemetryProperties();
-	assert(typeof telemetryProps.errorDetails === "string", "Error details should be present");
-	const detailedProperties = JSON.parse(
-		telemetryProps.errorDetails,
-	) as ITelemetryBaseProperties;
+	assert(isLayerIncompatibilityError(error), "Error should be a layerIncompatibilityError");
+	assert(typeof error.details === "string", "Error details should be present");
+	const detailedProperties = JSON.parse(error.details) as ITelemetryBaseProperties;
 	assert.strictEqual(
 		detailedProperties.isGenerationCompatible,
 		isGenerationCompatible,
 		"Generation compatibility not as expected",
 	);
+
+	assert.strictEqual(error.layer, "loader", "Layer type not as expected");
 	assert.strictEqual(
-		telemetryProps.loaderVersion,
-		pkgVersion,
-		"Loader version not as expected",
+		error.incompatibleLayer,
+		incompatibleLayer,
+		"Incompatible layer type not as expected",
 	);
+
+	assert.strictEqual(error.layerVersion, pkgVersion, "Loader version not as expected");
 	assert.strictEqual(
-		detailedProperties.loaderGeneration,
+		detailedProperties.layerGeneration,
```

Comment:
By putting the generation `details`, you're saying you don't really expect consumers to reason over it, but maybe to log it as part of this property bag?  Seems fine - you can always promote it to the interface later if needed.